### PR TITLE
removed support for the old 1.1 format and endpoints for rules stored in redis

### PR DESCRIPTION
**Pull request by lsmith77**
this also cleans up the code in other ways:

* merged several of the rule functions to reduce code duplication
* removed redundant code communicating with redis
* aligned function names
  * `fetch` is used for internal functions (so that `get`, `delete`, `post` etc can be used for routes)
  * `_from_redis` reads from redis
  * `_from_request` uses the request data to then read from relevant data sources (ie. redis)

also dropped the `/form` endpoint since it is for quite some time no longer used and redirect to `/editor` and not `/form`

### CODE REVIEW

**Review by hellasol — APPROVED** [review]

File: app/main.py [hunk 1]
```diff
@@ -1593,7 +1504,7 @@ def english_rules(version: float, config: Config, lang: Language, tokens, text:
         )
 
     if is_sub_category_enabled(config, "inclusive"):
-        list_full += rules_based_words_phrase_matcher_no_alt_en(
+        list_full += rules_based_words_phrase_matcher_en(
```
Comment (lnazarenko) [inline app/main.py]: rules_based_words_phrase_matcher_no_alt_en doesn't provide alternatives, currently it is used only for inclusive category
rules_based_words_phrase_matcher_en this is common function for English rules and it provides alternatives

File: app/main.py [hunk 2]
```diff
@@ -1857,7 +1772,8 @@ def plural_or_singular_en(
 ):
     if token_morph_number[0] == "Sing":
         return alternative_sing, subcategory
-    elif token_morph_number[0] == "Plur":
+
```
Comment (lnazarenko) [inline app/main.py:1833]: why did you change elif to if? Usually in Python if.. elif... else... statements

Comment (lsmith77) [reply app/main.py:1833]: because the previous `if` does a return

File: app/main.py (hunk 1 above)
Comment (lsmith77) [reply app/main.py]: yeah .. I have adapted `rules_based_words_phrase_matcher_en` to be able to handle the case when no alternatives are provided. I can show you how this works on Thursday

File: app/main.py [hunk 3]
```diff
@@ -2407,11 +2331,10 @@ def rules_based_words_phrase_matcher_de(
 
 # matcher to false positives
 def is_false_positive_match(list_false_positive, tokens, token):
-    if list_false_positive.__len__() > 0:
```
Comment (lnazarenko) [inline app/main.py:2404]: why did you remove the check of the length of false positives list?

**Review by lnazarenko — APPROVED** [review]

File: app/main.py (hunk 3 above)
Comment (lsmith77) [reply app/main.py:2404]: because if the list is empty, then the for loop will just not iterate over anything
